packages/webviz-core/src/dataProviders/MemoryCacheDataProvider.js

```diff
@@ -129,6 +134,9 @@ export default class MemoryCacheDataProvider implements DataProvider {
   // Total length of the data in nanoseconds. Used to compute progress with.
   _totalNs: number;
 
+  // Block index of the last read request. Used as a heuristic for a good place to prefetch data.
+  _cursorPosition: number;
```

Comment:
I think this is a duplicate of `_lastResolvedCallbackEnd`? Though I like `_cursorPosition` better as a name. 😄

Reply:
Done, thanks.

Reply:
Oh no I guess it's subtly different; you set `_cursorPosition` to the start of the range when adding a new read request, and we set `_lastResolvedCallbackEnd` to the end of a range when resolving a read request. Still, feels like we're using them for similar things…

Reply:
Yeah, they're different, but from playing around with it it works fine. I tried some other things and they had problems, but `_lastResolvedCallbackEnd` does the right thing.